Mathlib/FieldTheory/IntermediateField/Adjoin/Basic.lean

```diff
@@ -771,4 +771,40 @@ theorem cardinalMk_adjoin_le {E : Type u} [Field E] [Algebra F E] (s : Set E) :
     #(adjoin F s) ≤ #F ⊔ #s ⊔ ℵ₀ := by
   simpa using lift_cardinalMk_adjoin_le F s
 
+section AdjoinDouble
+
+variable {K L : Type*} [Field K] [Field L] [Algebra K L] {x y : L}
+
+theorem AdjoinDouble.isAlgebraic (hx : IsAlgebraic K x) (hy : IsAlgebraic K y) :
+    Algebra.IsAlgebraic K K⟮x, y⟯ := by
+  apply IntermediateField.isAlgebraic_adjoin
+  simp [hx.isIntegral, hy.isIntegral]
+
+theorem AdjoinDouble.finiteDimensional (hx : IsIntegral K x) (hy : IsIntegral K y) :
```

Comment:
Why use IsIntegral here and IsAlgebraic in the previous lemma? Over a field K they're equivalent.

Reply:
I switched to `IsIntegral` above.

Reply:
Personally I might actually be inclined to do it the other way since algebraic is a "weaker" hypothesis than integral?

Reply:
Also, this lemma could be called `finiteDimensional_adjoin_pair`.